config/buildspec-dev.yml

```diff
@@ -5,9 +5,9 @@ env:
     STAGE: dev
     SERVICE: exampleservice
     PROJECT: MozDef-EF
+    TYPE: <event-source-type>
```

Comment:
I think my only concern here is that it's not really an event source "type"
This has more to do with the method of obtaining events, so we could essentially just call this method-type, and perhaps instead of push/pull we could be more specific with "webhook" vs "rest"

Thoughts?

Reply:
everything else looks good, just concerned that event source type might be confusing.

Reply:
I got to thinking about this, and since we don't have a rest endpoint yet, this isn't something we really need to solve for right now, it's good to have, but could potentially just be set to webhook for now, and offer up the option of entering in the method when we do add different methods. Food for thought.

Reply:
I am OK with calling it something else other than "type". I am also OK to set the default value on it in buildspec. 

The potential issue with using webhook vs. rest/api as values is that, if we want to bake this parameter into the serverless stack name (which we do in this PR), then the serverless parameter bakes it into the serverless deployment bucket (the bucket used internally by the serverless framework to store the lambda functions etc.). This produces something like `<service>-webhook-dev-serverlessdeploymentbucket-j0d0dqrpg0n` which leaves 12 characters for the service name (as s3 bucket names can have up to 63 chars). The serverless stack won't deploy if this s3 bucket name exceeds the limit. I also explained in my previous comment that for some reason it is not easy to control the name of this bucket used internally by the serverless framework.

I was basically trying to maximise the space for service name. If we use pull/push it gives us a bit more wiggle room. But maybe max 12 chars for the service name is OK (the case when we use `webhook`).